quickwit-swim/src/membership.rs

```diff
@@ -119,7 +122,24 @@ impl ArtilleryMemberList {
         (suspect_members, down_members)
     }
 
-    pub fn mark_node_alive(&mut self, src_addr: &SocketAddr) -> Option<ArtilleryMember> {
+    // make sure node id is set correctly for the host (in case it changes)
+    pub fn ensure_node_id(&mut self, src_addr: SocketAddr, node_id: &str) {
```

Comment:
Can this method be called with our own `src_addr` and do what happens if it happens?

Reply:
yes, this can happen currently. We already have self member set with `remote_host: None`.

the "remove self duplicates" block removed the duplicate, but I will add a guard here to avoid this.